tester/src/main.rs

```diff
@@ -0,0 +1,136 @@
+    let client_options = ClientOptions::new()
+        .with_node("http://127.0.0.1:14265")?
+        .with_node_sync_disabled();
+    let account_manager = AccountManager::builder()
+        .with_secret_manager(secret_manager)
+        .with_client_options(client_options)
+        .with_coin_type(IOTA_COIN_TYPE)
+        .finish()
+        .await?;
+
+    Ok(account_manager)
+}
+
+async fn faucet<'a>(mnemonic: String) -> Result<(AccountManager, AccountHandle), Error> {
+    let faucet_manager = account_manager(Some(mnemonic)).await?;
+    faucet_manager.create_account().finish().await?;
+    let faucet_account = &faucet_manager.get_accounts().await?[0];
+
+    faucet_account.sync(None).await?;
+
+    Ok((faucet_manager, faucet_account.clone()))
+}
+
+async fn process_json<'a>(context: &Context<'a>, json: Value) -> Result<(), Error> {
+    if let Some(accounts) = json.get("accounts") {
+        process_accounts(context, accounts).await?;
+    }
+
+    if let Some(steps) = json.get("steps") {
+        process_steps(context, steps).await?;
+    }
+
+    if let Some(checks) = json.get("checks") {
+        process_checks(context, checks).await?;
+    }
+
+    Ok(())
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    let args: Vec<String> = std::env::args().collect();
+
+    let mnemonic = String::from("average day true meadow dawn pistol near vicious have ordinary sting fetch mobile month ladder explain tornado curious energy orange belt glue surge urban");
+    let (faucet_manager, faucet_account) = faucet(mnemonic).await?;
+    let protocol_parameters = faucet_account.client().get_protocol_parameters()?;
+
+    logger_init()?;
+
+    let mut paths = Vec::<PathBuf>::new();
+
+    if let Some(path) = args.get(1) {
+        paths.push(PathBuf::from(path));
+    } else {
+        let mut dir = fs::read_dir("json").await?;
+
+        while let Some(entry) = dir.next_entry().await? {
+            paths.push(entry.path());
+        }
+    };
+
+    for (index, path) in paths.iter().enumerate() {
+        let account_manager = account_manager(None).await?;
+
+        let context = Context {
+            faucet_manager: &faucet_manager,
+            faucet_account: &faucet_account,
+            account_manager,
+            protocol_parameters: protocol_parameters.clone(),
+        };
+
+        let content = fs::read_to_string(path).await?;
+        let json: Value = serde_json::from_str(&content)?;
+
+        log::info!(
+            "Executing test {}/{}: {:?}.",
+            index + 1,
+            paths.len(),
+            path.file_name().unwrap()
+        );
+        log::debug!("{}", json);
+
+        if let Err(err) = process_json(&context, json).await {
+            log::error!(
+                "Executing test {}/{}: {:?} failed: {}.",
+                index + 1,
+                paths.len(),
+                path.file_name().unwrap(),
+                err
+            );
+        }
+    }
```

Comment:
After all tests, if there was an error, we need to add an exit code so the CI fails

Reply:
Will do